h5pt.go

```diff
@@ -12,32 +12,46 @@ package hdf5
 import "C"
 
 import (
+	"encoding/binary"
 	"fmt"
+	"math"
 	"reflect"
 	"runtime"
 	"unsafe"
 )
 
+var nativeEndian binary.ByteOrder
+
+func init() {
+	one := uint16(0x1)
+	nativeEndian = [...]binary.ByteOrder{
+		binary.BigEndian,
+		binary.LittleEndian,
+	}[(*[2]byte)(unsafe.Pointer(&one))[0]]
+}
+
 // appendData is a wrapper type for information necessary to create and
 // subsequently write an in memory object to a PacketTable using Append().
 type appendData struct {
-	ptr        unsafe.Pointer
-	memSize    C.size_t
-	offset     C.size_t
-	numRecords C.size_t
+	buf        []byte
```

Comment:
it dawns on me that now, `appendData` looks more and more like an encoder, pretty much like `encoding/json.Encoder`.

`extractValues` could be a method on that `encoder` type:
```go
func (enc *encoder) encode(v interface{}) error { ... }
```